zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,201 @@
+        this.connectionStr = null;
+        this.zNodePath = requireNonNull(zNodePath, "zNodePath");
+        checkArgument(!this.zNodePath.isEmpty(), "zNodePath can't be empty");
+    }
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a ZooKeeper connection string and a zNode path.
+     *
+     * @param connectionStr the ZooKeeper connection string
+     * @param zNodePath the ZooKeeper node to register
+     */
+    public ZooKeeperUpdatingListenerBuilder(String connectionStr, String zNodePath) {
+        this.connectionStr = requireNonNull(connectionStr, "connectionStr");
+        checkArgument(!this.connectionStr.isEmpty(), "connectionStr can't be empty");
+        this.zNodePath = requireNonNull(zNodePath, "zNodePath");
+        checkArgument(!this.zNodePath.isEmpty(), "zNodePath can't be empty");
+    }
+
+    private void ensureInternalClient() {
+        checkState(client == null,
+                   "This method is allowed only when created with a connection string.");
+    }
+
+    /**
+     * Sets the connect timeout.
+     *
+     * @param connectTimeout the connect timeout
+     *
+     * @throws IllegalStateException if this builder is constructed with
+     *                               {@link #ZooKeeperUpdatingListenerBuilder(CuratorFramework, String)}
+     */
+    public ZooKeeperUpdatingListenerBuilder connectTimeout(Duration connectTimeout) {
+        requireNonNull(connectTimeout, "connectTimeout");
+        checkArgument(!connectTimeout.isZero() && !connectTimeout.isNegative(),
+                      "connectTimeout: %s (expected: > 0)", connectTimeout);
+        return connectTimeoutMillis(connectTimeout.toMillis());
+    }
+
+    /**
+     * Sets the connect timeout (in ms). (default: 1000)
+     *
+     * @param connectTimeoutMillis the connect timeout
+     *
+     * @throws IllegalStateException if this builder is constructed with
+     *                               {@link #ZooKeeperUpdatingListenerBuilder(CuratorFramework, String)}
+     */
+    public ZooKeeperUpdatingListenerBuilder connectTimeoutMillis(long connectTimeoutMillis) {
+        ensureInternalClient();
+        checkArgument(connectTimeoutMillis > 0,
+                      "connectTimeoutMillis: %s (expected: > 0)", connectTimeoutMillis);
+        this.connectTimeoutMillis = (int) Math.min(connectTimeoutMillis, Integer.MAX_VALUE);
+        return this;
+    }
+
+    /**
+     * Sets the session timeout.
+     *
+     * @param sessionTimeout the session timeout
+     *
+     * @throws IllegalStateException if this builder is constructed with
+     *                               {@link #ZooKeeperUpdatingListenerBuilder(CuratorFramework, String)}
+     */
+    public ZooKeeperUpdatingListenerBuilder sessionTimeout(Duration sessionTimeout) {
+        requireNonNull(sessionTimeout, "sessionTimeout");
+        checkArgument(!sessionTimeout.isZero() && !sessionTimeout.isNegative(),
```

Comment:
How about removing this line?

Reply:
That would result in incorrect exception message. `sessionTimeout: ...` vs `sessionTimeoutMillis: ...`